apps/emqx_machine/src/emqx_machine.erl

```diff
@@ -94,3 +98,89 @@ configure_shard_transports() ->
         end,
         maps:to_list(ShardTransports)
     ).
+
+set_mnesia_extra_diagnostic_checks() ->
+    Checks = [{check_open_ports, ok, fun ?MODULE:open_ports_check/0}],
+    mria_config:set_extra_mnesia_diagnostic_checks(Checks),
+    ok.
+
+-define(DEFAULT_EKKA_PORT, 4370).
+-define(PORT_PROBE_TIMEOUT, 10_000).
+open_ports_check() ->
+    %% expected nodes according to mnesia schema
+    OtherNodes = mnesia:system_info(db_nodes) -- [node()],
```

Comment:
- [ ] todo: fix for replicants

Reply:
thanks for the info, thought it was a manual call by human.